tsl/test/expected/attach_chunk.out

```diff
@@ -0,0 +1,231 @@
+SELECT count(*) FROM attach_test WHERE time > '2025-07-01'::timestamptz;
+ count 
+-------
+     1
+(1 row)
+
+INSERT INTO attach_test VALUES (5, '2025-07-15 05:00:00-8', 2, 19.5);
+SELECT count(*) FROM regular_table_to_attach;
+ count 
+-------
+     2
+(1 row)
+
+-- Detach and re-attach a chunk
+CALL detach_chunk(:'CHUNK_NAME');
+CALL attach_chunk('attach_test', :'CHUNK_NAME', :'CHUNK_SLICES');
+-- Verify it's re-attached
+SELECT count(*) > 0 FROM pg_inherits WHERE inhrelid = :'CHUNK_NAME'::regclass::oid;
+ ?column? 
+----------
+ t
+(1 row)
+
+-- Verify constraints and indexes are restored
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE chunk_id = :'CHUNK_ID';
+ chunk_id | dimension_slice_id | constraint_name | hypertable_constraint_name 
+----------+--------------------+-----------------+----------------------------
+(0 rows)
+
+SELECT * FROM _timescaledb_catalog.chunk_index WHERE chunk_id = :'CHUNK_ID';
+ chunk_id | index_name | hypertable_id | hypertable_index_name 
+----------+------------+---------------+-----------------------
+(0 rows)
+
+SELECT * FROM _timescaledb_catalog.dimension_slice ds
+JOIN _timescaledb_catalog.chunk_constraint cc ON ds.id = cc.dimension_slice_id
+WHERE cc.chunk_id = :'CHUNK_ID';
+ id | dimension_id | range_start | range_end | chunk_id | dimension_slice_id | constraint_name | hypertable_constraint_name 
+----+--------------+-------------+-----------+----------+--------------------+-----------------+----------------------------
+(0 rows)
+
+-- Attach a chunk to another hypertable with a different dimension
+CALL detach_chunk('regular_table_to_attach');
+CREATE TABLE hypertable_with_different_dimension(id int, time timestamptz not null, device int, temp float);
+SELECT * FROM create_hypertable('hypertable_with_different_dimension', 'time');
+ hypertable_id | schema_name |             table_name              | created 
+---------------+-------------+-------------------------------------+---------
+             2 | public      | hypertable_with_different_dimension | t
+(1 row)
+
+CALL attach_chunk('hypertable_with_different_dimension', 'regular_table_to_attach', '{"time": [1752105600000000, 1752710400000000]}');
+CALL detach_chunk('regular_table_to_attach');
+CREATE TABLE not_a_hypertable(id int, time timestamptz not null, device int, temp float);
+-- Error cases
+\set ON_ERROR_STOP 0
+CALL attach_chunk('attach_test', 'nonexistent_table', '{"time": [1752105600000000, 1752710400000000], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  relation "nonexistent_table" does not exist at character 34
+CALL attach_chunk('nonexistent_hypertable', 'regular_table_to_attach', '{"time": [1752105600000000, 1752710400000000], "id": [-9223372036854775808, 1073741823]}');
+ERROR:  relation "nonexistent_hypertable" does not exist at character 19
+CALL attach_chunk('not_a_hypertable', 'regular_table_to_attach', '{"time": [1752105600000000, 1752710400000000], "id": [-9223372036854775808, 1073741823]}');
```

Comment:
Make sure you also test  "bad" json. And good json, but bad/weird info/constraints.